include/swift/SIL/SILLocation.h

```diff
@@ -84,10 +84,15 @@ class SILLocation {
     DebugInfoKind = 1 << 3 | 1 << 4
   };
 
-  typedef struct {
-    unsigned Line = 0, Column = 0;
-    const char *Filename = nullptr;
-  } DebugLoc;
+  struct DebugLoc {
+    unsigned Line;
+    unsigned Column;
+    const char *Filename;
+
+    DebugLoc(unsigned Line = 0, unsigned Column = 0,
+             const char *Filename = nullptr) : Line(Line), Column(Column),
+                                               Filename(Filename) { }
+  };
```

Comment:
> On Mar 18, 2016, at 3:03 PM, eeckstein <notifications@github.com> wrote:
> 
> In include/swift/SIL/SILLocation.h <https://github.com/apple/swift/pull/1735#discussion_r56728646>:
> 
> > @@ -84,10 +84,15 @@ class SILLocation {
> >      DebugInfoKind = 1 << 3 | 1 << 4
> >    };
> >  
> > -  typedef struct {
> > -    unsigned Line = 0, Column = 0;
> > -    const char *Filename = nullptr;
> > -  } DebugLoc;
> > +  struct DebugLoc {
> > +    unsigned Line;
> > +    unsigned Column;
> > +    const char *Filename;
> > +
> > +    DebugLoc(unsigned Line = 0, unsigned Column = 0,
> > +             const char *Filename = nullptr) : Line(Line), Column(Column),
> > +                                               Filename(Filename) { }
> > +  };
> Existing code in bool SILParser::parseSILLocation(SILLocation &Loc).
> It's probably better to rewrite this and not allow an empty DebugLoc.
> 
> 
Ah yes. But rewriting involves a tedious rewriting of all call sites. We don’t necessarily need to rewrite parseSILLocation() now, but it’s probably better to not add this constructor so we don’t encourage creation of empty locations.

-- adrian